### DISCUSSION TITLE
n_process > 1 throws CUDA error while using en_core_web_trf

### DISCUSSION THREAD

**Discussion by kaushikepi** [explosion/spaCy#9022] [Help: Coding & Implementations]
How can we use multiprocessing in the spacy v3 en_core_web_trf model using GPU?

There is no clear documentation or issue.

Can anyone share some insight or suggest some solution?

@honnibal @ines

**Comment by kaushikepi** [comment]
@svlandeg Can you share some insight on it?

**Comment by svlandeg** [comment]
Hi,

This is a good topic for the discussion forum so I'll move it there. I also want to ask you to refrain from pinging various maintainers, as this makes our mentions unusable. We provide support here on a best-effort basis and will get to your question in due time. Thanks for understanding!

**Comment by DavidGerva** [comment]
Hi @kaushikepi,
you provide few information, so I don't know on which platforms or services you're trying to use the GPUs. In any case I used that model with GPUs on gcloud and having encountered some difficulties I wrote this article with a step by step instructions:
https://dev.to/davidgerva/spacy-3-on-a-google-cloud-compute-instance-to-train-a-ner-transformer-model-23hf

I don't know, but maybe it can help?!

EDIT: I'm just reading now you're speaking about "multiprocessing": my tutorial is not about/with multiprocessing. If you exactly want that, maybe it can be not so useful.

**Comment by kaushikepi** [comment]
Hey @DavidGerva 
Thanks for your reply.

I have been using AWS to spawn an Ec2 instance with Deep Learning AMI. I am able to use GPU with spacy transformer model for inference but when adding the `n_process` parameter with `value > 1 `then it throws an error and keeps running infinitely.

System Configuration
Cuda v10.0

Dockerfile used:
```
FROM nvidia/cuda:10.0-devel-ubuntu18.04

ENV LC_ALL=C.UTF-8
ENV LANG=C.UTF-8

# Install base packages.
RUN apt-get update --fix-missing && apt-get install -y \
    bzip2 \
    ca-certificates \
    python3-dev \
    python3-pip \
    libc-dev \
    libglib2.0-0 \
    libsm6 \
    libxext6 \
    libxrender1 \
    libevent-dev \
    libgl1-mesa-glx \
    build-essential && \
    rm -rf /var/lib/apt/lists/*

ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64

# Tell nvidia-docker the driver spec that we need as well as to
# use all available devices, which are mounted at /usr/local/nvidia.
# The LABEL supports an older version of nvidia-docker, the env
# variables a newer one.
ENV NVIDIA_VISIBLE_DEVICES all
ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
LABEL com.nvidia.volumes.needed="nvidia_driver"

#Install spacy and its dependencies
RUN pip3 install --upgrade pip && \
    pip3 install -U setuptools wheel && \
    pip3 install spacy==3.0.5 spacy-transformers==1.0.1 cupy-cuda100 && \
    python3 -m spacy download en_core_web_trf

#Install server packages
RUN pip3 install fastapi uvicorn pydantic

WORKDIR /app
COPY ./app /app

EXPOSE 5000

ENTRYPOINT ["uvicorn"]
CMD ["api:app", "--host", "0.0.0.0", "--timeout-keep-alive", "300", "--port", "5000"]
```

**Reply by DavidGerva** [reply]
I'm not sure I'll be of help, but can you please post the error you get?
